extensions/stackdriver/common/utils.cc

```diff
@@ -35,6 +36,23 @@ const std::string getContainerName(
   return kIstioProxyContainerName;
 }
 
+const std::string getNodeID(
+    const flatbuffers::Vector<flatbuffers::Offset<flatbuffers::String>>
+        *ip_addrs) {
+  if (!ip_addrs || ip_addrs->size() == 0) {
+    return "istio-proxy";
+  }
+
+  std::string node_id;
+  auto it = ip_addrs->begin();
+  node_id = flatbuffers::GetString(*it);
+  ++it;
+  for (; it != ip_addrs->end(); ++it) {
+    node_id = absl::StrCat(node_id, ",", flatbuffers::GetString(*it));
```

Comment:
nit: StrJoin is probably faster since you are allocating a string each time here.

Reply:
do mean instead of the loop?  or instead of `StrCat` inside of the loop?  `StrJoin` had issues taking ip_addrs directly, iirc.

Reply:
done. ptal.